libcst/codemod/_cli.py

```diff
@@ -250,10 +250,13 @@ def _execute_transform(  # noqa: C901
         # We do this after the fork so that a context that was initialized with
         # some defaults before calling parallel_exec_transform_with_prettyprint
         # will be updated per-file.
+        # Clean the warnings as well, otherwise they will be
+        # passed from the previous file
         transformer.context = replace(
             transformer.context,
             filename=filename,
             scratch=deepcopy(scratch),
+            warnings=[],
```

Comment:
While this change will fix the issue, there's a deeper problem here. Apart from `metadata_manager`, every other field of `context` should be reset per file.
Concretely, I'm surprised we change `filename` here but not `full_module_name` or `full_package_name`.

Maybe we should make this explicit and have a global context that persists between files, and a local one that gets reset. Then this assignment would look something like:

```python
try:
    mod_and_pkg = calculate_module_and_package(config.repo_root or ".", filename)
    mod_name = mod_and_pkg.name
    pkg_name = mod_and_pkg.package
except ValueError as exc:
    print(f"Failed to determine module name for {filename}: {ex}", file=sys.stderr)
    mod_name = None
    pkg_name = None
    
transformer.context = CodemodContext(scratch=deepcopy(scratch), filename=filename, full_module_name=mod_name, full_package_name=pkg_name, metadata_manager=transformer.context.metadata_manager)

Reply:
Having said that, I'm happy to merge the PR as is because it's definitely an improvement over the current state. Let me know if you wanna roll this change into this PR or not

Reply:
`full_module_name` or `full_package_name` are changed below, so currently context replaced twice. And if it isn't possible to determine the module/package name, it will persist from the previous file. The `wrapper` also persists. That would be changing in the new commit I'm submitting.

**Before:**
- `full_module_name`/`full_package_name` persist if can't get them for the new file
- `wrapper` persists across files

**After:**
- `full_module_name`/`full_package_name` both set to `None` if can't figure them out.
- `wrapper`  is set to `None` on every file.

Is that okay to not set the `wrapper`?

Reply:
Yep, it should be